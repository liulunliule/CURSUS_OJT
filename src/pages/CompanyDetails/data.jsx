import { useState, useEffect } from "react";
import axios from "axios";

export const useFetchOffices = () => {
  const [offices, setOffices] = useState([]);

  useEffect(() => {
    axios.get("https://668f4a0880b313ba09178dee.mockapi.io/api/offices")
      .then(response => {
        setOffices(response.data);
      })
      .catch(error => {
        console.error("Error fetching office data:", error);
      });
  }, []);

  return offices;
};