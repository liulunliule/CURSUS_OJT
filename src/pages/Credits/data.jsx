import { useState, useEffect } from "react";
import axios from "axios";
import {
    fetchCreditData,
    fetchTransactionsData,
} from "../../services/apiService";

export const useFetchCreditsData = () => {
    const [addedCredits, setAddedCredits] = useState({});
    const [transactions, setTransactions] = useState([]);

    useEffect(() => {
        const fetchAddedCredits = async () => {
            try {
                const response = await fetchCreditData();
                setAddedCredits(response.data[0]);
            } catch (error) {
                console.error("Error fetching added credits data:", error);
            }
        };

        const fetchTransactions = async () => {
            try {
                const response = await fetchTransactionsData();
                setTransactions(response.data);
            } catch (error) {
                console.error("Error fetching transactions data:", error);
            }
        };

        fetchAddedCredits();
        fetchTransactions();
    }, []);

    return { addedCredits, transactions };
};
