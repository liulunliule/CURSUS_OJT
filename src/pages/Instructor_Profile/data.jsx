import axios from "axios";

export const fetchUserInfo = async (targetId) => {
  try {
    const response = await axios.get("https://66751909a8d2b4d072eeb172.mockapi.io/InstructorProfile");
    return response.data.find((item) => item.id === targetId);
  } catch (error) {
    console.error("Error fetching user info:", error);
    throw error;
  }
};

export const fetchUserPosts = async () => {
  try {
    const response = await axios.get("https://6677cf6e0bd45250561c9488.mockapi.io/course");
    return response.data;
  } catch (error) {
    console.error("Error fetching user posts:", error);
    throw error;
  }
};

export const fetchUserReviews = async () => {
  try {
    const response = await axios.get("https://6677cf6e0bd45250561c9488.mockapi.io/reviews");
    return response.data;
  } catch (error) {
    console.error("Error fetching user reviews:", error);
    throw error;
  }
};

export const fetchSubscriptions = async () => {
  try {
    const response = await axios.get("https://6678f8ae0bd4525056207d59.mockapi.io/subscriptions");
    return response.data;
  } catch (error) {
    console.error("Error fetching subscriptions:", error);
    throw error;
  }
};
