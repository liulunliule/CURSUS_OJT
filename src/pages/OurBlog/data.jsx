import { useState, useEffect } from "react";
import axios from "axios";

export const useFetchBlogs = () => {
  const [blogs, setBlogs] = useState([]);

  useEffect(() => {
    axios.get("https://668f4a0880b313ba09178dee.mockapi.io/api/blogs")
      .then((response) => setBlogs(response.data))
      .catch((error) => console.error("Error fetching blog data:", error));
  }, []);

  return blogs;
};