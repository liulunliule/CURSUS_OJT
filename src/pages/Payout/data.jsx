import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchNextPayout, fetchTransactions } from '../../redux/features/payoutSlice';

export const useFetchPayoutData = () => {
  const dispatch = useDispatch();
  const nextPayout = useSelector((state) => state.payout.nextPayout);
  const transactions = useSelector((state) => state.payout.transactions);
  const status = useSelector((state) => state.payout.status);
  const error = useSelector((state) => state.payout.error);

  useEffect(() => {
    if (status === 'idle') {
      dispatch(fetchNextPayout());
      dispatch(fetchTransactions());
    }
  }, [status, dispatch]);

  return { nextPayout, transactions, status, error };
};