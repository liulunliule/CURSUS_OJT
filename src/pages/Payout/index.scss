.payout-page {
  display: flex;
  font-family: Arial, sans-serif;
  margin-top: 50px;
  margin-left: 240px;
  transition: margin-left 0.5s ease;

    &.active{
        transition: margin-left 0.5s ease;
        margin-left: 0;
    }

  .content {
    flex: 1;
    padding: 20px;

    .payout-header {
      h2 {
        font-size: 20px;
        margin-bottom: 20px;
        align-items: flex-start;
      }
    }

    .payout-content {
      display: flex;
      flex-direction: column;
      margin-top: 50px;

      .next-payout {
        width: 377.4px;
        height: 206.35px;
        background-color: #fff;
        padding: 20px;
        border: 1px solid #ddd;
        border-radius: 5px;
        margin-bottom: 20px;
        align-items: flex-start;

        h3 {
          margin-top: 0;
          margin-bottom: 20px;
        }

        p {
          margin: 5px 0;

          span {
            font-weight: bold;
            color: gray;
            font-size: 14px;
          }
        }

        h6 {
          margin: 5px 0;
          font-weight: bold;
          color: rgb(175, 174, 174);
          font-size: 25px;
        }

        h5 {
          margin: 5px 0;
          font-weight: bold;
          color: gray;
          font-size: 14px;
          margin-bottom: 20px;
        }
      }

      .payout-account {
        width: 377.4px;
        background-color: #fff;
        padding: 20px;
        border: 1px solid #ddd;
        border-radius: 5px;
        display: flex;
        flex-direction: column;
        align-items: flex-start;

        h3 {
          margin-top: 0;
          margin-bottom: 20px;
        }

        .account-details {
          display: flex;
          align-items: center;
          margin-bottom: 20px;

          img {
            margin-right: 20px;
            width: 80px;
            height: 28px;
          }

          p {
            flex: 1;
          }
        }

        button {
          width: 335px;
          padding: 10px 20px;
          background-color: red;
          color: #fff;
          border: none;
          border-radius: 5px;
          cursor: pointer;
          align-self: flex-end;
          margin-top: auto;

          &:hover {
            background-color: black;
          }
        }
      }
    }

    .transactions {
      width: 730px;
      margin-top: 50px;
      margin-right: 30px;

      table {
        width: 100%;
        border-collapse: collapse;

        thead {
          tr {
            th {
              padding: 10px;
              border-bottom: 1px solid #ddd;
              text-align: left;
              background-color: #ffe3e3;
            }
          }
        }

        tbody {
          tr {
            td {
              padding: 10px;
              border-bottom: 1px solid #ddd;
              background-color: #fff;
            }
          }
        }
      }
    }
  }
}