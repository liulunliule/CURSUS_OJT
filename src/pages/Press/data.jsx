import { useState, useEffect } from "react";
import axios from "axios";

export const useFetchPressData = () => {
  const [news, setNews] = useState([]);
  const [releases, setReleases] = useState([]);

  useEffect(() => {
    const fetchNews = async () => {
      try {
        const response = await axios.get('https://6690833cc0a7969efd9c4da4.mockapi.io/api/press_news');
        setNews(response.data);
      } catch (error) {
        console.error("Error fetching news data:", error);
      }
    };

    const fetchReleases = async () => {
      try {
        const response = await axios.get('https://6690833cc0a7969efd9c4da4.mockapi.io/api/press_releases');
        setReleases(response.data);
      } catch (error) {
        console.error("Error fetching releases data:", error);
      }
    };

    fetchNews();
    fetchReleases();
  }, []);

  return { news, releases };
};