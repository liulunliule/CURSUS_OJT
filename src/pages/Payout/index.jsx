import React from 'react';
import './index.scss';
import { payout } from "../../assets";
import { useSelector } from 'react-redux';
import { useFetchPayoutData } from './data';

const Payout = () => {
  const isShowAll = useSelector((state) => state.savedCourse.isShowAll);
  const { nextPayout, transactions } = useFetchPayoutData();

  return (
    <div className={`payout-page ${isShowAll ? "active" : ""}`}>
      <div className="content">
        <div className="payout-header">
          <h2>Payout</h2>
        </div>
        <div className="payout-content">
          <div className="next-payout">
            <h3>Next payout</h3>
            <h6>{nextPayout.amount}</h6>
            <h5>via {nextPayout.method}</h5>
            <p>Your payout will be processed on <span>{nextPayout.date}</span></p>
          </div>
          <div className="payout-account">
            <h3>Payout account</h3>
            <div className="account-details">
              <img src={payout} alt="Payout" />
              <p>Added: 01 Mar 2020</p>
            </div>
            <button>Set Account</button>
          </div>
        </div>
        <div className="transactions">
          <table>
            <thead>
              <tr>
                <th>Amount</th>
                <th>Payout Method</th>
                <th>Date Processed</th>
              </tr>
            </thead>
            <tbody>
              {transactions.map((transaction, index) => (
                <tr key={index}>
                  <td>{transaction.amount}</td>
                  <td>{transaction.method}</td>
                  <td>{transaction.date}</td>
                </tr>
              ))}
            </tbody>
          </table>
        </div>
      </div>
    </div>
  );
};

export default Payout;