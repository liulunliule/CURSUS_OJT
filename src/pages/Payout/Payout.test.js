import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { useSelector } from 'react-redux';
import Payout from './index';
import { useFetchPayoutData } from './data';

jest.mock('react-redux', () => ({
  useSelector: jest.fn(),
}));

jest.mock('./data', () => ({
  useFetchPayoutData: jest.fn(),
}));

describe('Payout Component', () => {
  const mockData = {
    nextPayout: { amount: '$100', method: 'Credit Card', date: '2023-07-27' },
    transactions: [
      { amount: '$100', method: 'Credit Card', date: '2023-07-27' },
      { amount: '$50', method: 'PayPal', date: '2023-07-28' },
    ],
    status: 'succeeded',
    error: null,
  };

  beforeEach(() => {
    useSelector.mockClear();
    useFetchPayoutData.mockClear();
  });

  it('renders loading state', () => {
    useFetchPayoutData.mockReturnValue({ status: 'loading' });
    render(<Payout />);
    expect(screen.getByText(/Loading.../i)).toBeInTheDocument();
  });

  it('renders error state', () => {
    useFetchPayoutData.mockReturnValue({ ...mockData, status: 'failed', error: 'Failed to fetch data' });
    render(<Payout />);
    expect(screen.getByText(/Error: Failed to fetch data/i)).toBeInTheDocument();
  });

  it('renders payout data correctly', () => {
    useFetchPayoutData.mockReturnValue(mockData);
    render(<Payout />);
    expect(screen.getByText(/Next payout/i)).toBeInTheDocument();
    expect(screen.getByText(/via Credit Card/i)).toBeInTheDocument();
    expect(screen.getByText(/Your payout will be processed on/i)).toBeInTheDocument();
  });

  it('applies active class when isShowAll is true', () => {
    useFetchPayoutData.mockReturnValue(mockData);
    useSelector.mockReturnValue(true);
  });

  it('does not apply active class when isShowAll is false', () => {
    useFetchPayoutData.mockReturnValue(mockData);
    useSelector.mockReturnValue(false);
  });
});