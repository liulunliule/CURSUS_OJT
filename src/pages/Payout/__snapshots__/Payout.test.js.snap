// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Payout 1`] = `
<div
  className="payout-page "
>
  <div
    className="content"
  >
    <div
      className="payout-header"
    >
      <h2>
        Payout
      </h2>
    </div>
    <div
      className="payout-content"
    >
      <div
        className="next-payout"
      >
        <h3>
          Next payout
        </h3>
        <h6 />
        <h5>
          via 
        </h5>
        <p>
          Your payout will be processed on 
          <span />
        </p>
      </div>
      <div
        className="payout-account"
      >
        <h3>
          Payout account
        </h3>
        <div
          className="account-details"
        >
          <img
            alt="Payout"
            src="payout.jpg"
          />
          <p>
            Added: 01 Mar 2020
          </p>
        </div>
        <button>
          Set Account
        </button>
      </div>
    </div>
    <div
      className="transactions"
    >
      <table>
        <thead>
          <tr>
            <th>
              Amount
            </th>
            <th>
              Payout Method
            </th>
            <th>
              Date Processed
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
    </div>
  </div>
</div>
`;
